<a href="https://colab.research.google.com/github/ChitikelaNeelaPadmavathi/python-codes/blob/main/DAY_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#1.palindrome

In [ ]:
s=int(input ())
reverse=int(str(s)[::-1])
if s==reverse:
    print ("palindrome")
else:
    print ("not a palindrome ")

34567
not a palindrome 


In [ ]:

def is_palindrome(string):
    string = string.lower().replace(" ", "")  # Convert to lowercase and remove spaces
    return string == string[::-1]  # Check if the processed string is the same when reversed

# Get user input
s = input("Enter a string: ")

# Check and print the result
if is_palindrome(s):
    print("Palindrome")
else:
    print("Not a palindrome")

Enter a string: Gdrhj
Not a palindrome


**2.AMSTRONG NUMBER**


In [ ]:
num=int(input())
digits=str(num)
num_digit=len(digits)
amstrong_sum=0
for i in digits:
  amstrong_sum+=int(i)**num_digit
if amstrong_sum==num:
    print("It is Amstrong")
else:
    print("It is not a amstrong number")



153
It is Amstrong


In [ ]:
n=int(input())
m=n
sum=0
while m!=0:
  d=m%10
  sum=sum+(d**3)
  m=m//10
if(sum==n):
  print("amstrong")
else:
  print("not amstrong")


5667
not amstrong


**3.FIBONACCI SERIES**

In [ ]:
num=int(input())
n1,n2=0,1
print("fibanacii:",n1,n2,end=" ")
for i in range(2,num):
  n3=n1+n2
  n1=n2
  n2=n3
  print(n3,end=" ")

10
fibanacii: 0 1 1 2 3 5 8 13 21 34 

**4.FACTORIAL NUMBER**

In [ ]:
num=int(input())
fact=1
for i in range(1,num+1):
  fact=fact*i
print(fact)

5
120


**5.POWER OF A NUMBER**

In [ ]:
num=int(input())
b=num*2
print(b)

5
10


**6.FACTORS OF A NUMBER**

In [ ]:
num=int(input())
for i in range(1,num+1):
  if num%i==0:
    print(i,end=" ")





8
1 2 4 8 

**7.Sum of prime factors**

In [ ]:
num = int(input("enter the num: "))
sum = 0
for i in range(1,10):
     if num % i == 0:
          sum = sum + i
print(sum)

enter the num: 10
8


**8.STRONG NUMBER**
Input : 145
Output : It's a Strong Number.
Explanation : Number = 145
145 = 1! + 4! + 5!
145 = 1 + 24 + 120

In [ ]:
import math
n=int(input())
m=n
sum=0
while m!=0:
  d=m%10
  sum=sum+(math.factorial(d))
  m=m//10
if(sum==n):
  print("strorng")
else:
  print("not strong")



145
strorng


****